### ISSUE TITLE
Bug: Custom validation error for a field with an asterisk.

### ISSUE THREAD

**Issue by iRedds** [codeigniter4/CodeIgniter4#6245]
### CodeIgniter4 Version

dev

### What happened?

If you specify a custom error message for a field with an asterisk, the default error text will be used.
This is due to the fact that a field without an asterisk is passed to generate an error (it contains the full path of the key)

### Steps to Reproduce

```php
$data = [
    'user' => [
        ['name' => null],
        ['name' => null],
    ]
];

$validator = Services::validation();

$validator->setRules(['user.*.name' => 'required'], ['user.*.name' => ['required' => 'Required']]);

$validator->run($data);

dd($validator->getErrors());
// array (2) [
//     'user.0.name' => 'The user.*.name field is required.'
//     'user.1.name' => 'The user.*.name field is required.'
// ]
```

### Expected Output

```php
array (2) [
    'user.0.name' => 'Required'
    'user.1.name' => 'Required'
]
```

**Comment by ping-yee** [comment]
I sent a PR, review please. #6340

**Comment by ping-yee** [comment]
I have been closed the PR #6340 , and sent the new one #6352 .
Review, please.
